fix(ios): avoid crash when openStream fails

sgtcoolguy opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-27515

**Description:**
wraps in try/catch to properly set exception on JS context
if JS code is in async callback/event/timer and has no try/catch it will still crash
Also, differs from 'old' proxies in that the uncaught exception handler code isn't invoked

I'm not sure what the correct way to handle this is. This at least offers the user the ability to avoid the crash. If the code is invoked in the main body of a js file, it'll pop the error handler/controller with the stack trace.

But if it's invoked on an event listener it will:
- crash if not surrounded by try/catch
- let the user handle the error if caught by try/catch

I don't think it's specific to this PR, but we should investigate having the "unhandled" case on an event listener/callback/timer also show the error controller like the "old proxies" (the ones using the JS C API, not the Obj-C API) do.

I thought we had tried to interject an unhandled exception handler in the Obj-C API before and had to roll it back because it was affecting the mocha test suite's timer behavior (See #10862)

sgtcoolguy committed:
fix(ios): avoid crash when openStream fails

wraps in try/catch to properly set exception on JS context
if JS code is in async callback/event/timer and has no try/catch it will still crash
Also, differs from 'old' procies in that the uncaught exception handler code isn't invoked

Fixes TIMOB-27515

build commented:
<!--
  2 failure:  :microscope: Ther..., Tests have failed...
  0 warning: 
  3 messages
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/sgtcoolguy/titanium_mobile/blob/TIMOB-27515/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-11308/1/artifact/dist/mobilesdk-8.3.0.v20191031121017-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:x: 20 tests have failed
There are 20 tests failing and 473 skipped out of 4741 total tests.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  </tbody>
</table>

### Tests:

<table><tr><th>Classname</th><th>Name</th><th>Time</th><th>Error</th></tr>
<tr><td>android.emulator.Titanium.Blob</td><td>.size returns byte count of non-image (JS file)</td><td>0.001</td><td><pre>Error: expected 0 to be above 0
at Assertion.fail (/node_modules/should/cjs/should.js:275:13)
      at Assertion.value (/node_modules/should/cjs/should.js:356:9)
      at Context.&lt;anonymous&gt; (/ti.blob.test.js:158:22)
      at Test.Runnable.run (/ti-mocha.js:6535:54163)
      at Runner.runTest (/ti-mocha.js:6535:58604)
      at /ti-mocha.js:6535:59445
      at next (/ti-mocha.js:6535:58013)
      at /ti-mocha.js:6535:57975
      at next (/ti-mocha.js:6535:57665)
      at /ti-mocha.js:6535:57759</pre></td></tr>
<tr><td>android.emulator.Titanium.Blob</td><td>#append() appends two files together</td><td>0.024</td><td><pre>Error: expected 91276 to equal 0
at Assertion.fail (/node_modules/should/cjs/should.js:275:13)
      at Assertion.value (/node_modules/should/cjs/should.js:356:9)
      at Context.&lt;anonymous&gt; (/ti.blob.test.js:114:28)
      at Test.Runnable.run (/ti-mocha.js:6535:54163)
      at Runner.runTest (/ti-mocha.js:6535:58604)
      at /ti-mocha.js:6535:59445
      at next (/ti-mocha.js:6535:58013)
      at /ti-mocha.js:6535:57975
      at next (/ti-mocha.js:6535:57665)
      at /ti-mocha.js:6535:57759</pre></td></tr>
<tr><td>android.emulator.Titanium.Blob</td><td>.length</td><td>0.004</td><td><pre>Error: expected 0 to be above 0
at Assertion.fail (/node_modules/should/cjs/should.js:275:13)
      at Assertion.value (/node_modules/should/cjs/should.js:356:9)
      at Context.&lt;anonymous&gt; (/ti.blob.test.js:149:24)
      at Test.Runnable.run (/ti-mocha.js:6535:54163)
      at Runner.runTest (/ti-mocha.js:6535:58604)
      at /ti-mocha.js:6535:59445
      at next (/ti-mocha.js:6535:58013)
      at /ti-mocha.js:6535:57975
      at next (/ti-mocha.js:6535:57665)
      at /ti-mocha.js:6535:57759</pre></td></tr>
<tr><td>android.emulator.Titanium.Filesystem.File</td><td>.size</td><td>0</td><td><pre>Error: expected 0 to be above 0
at Assertion.fail (/node_modules/should/cjs/should.js:275:13)
      at Assertion.value (/node_modules/should/cjs/should.js:356:9)
      at Context.&lt;anonymous&gt; (/ti.filesystem.file.test.js:91:22)
      at Test.Runnable.run (/ti-mocha.js:6535:54163)
      at Runner.runTest (/ti-mocha.js:6535:58604)
      at /ti-mocha.js:6535:59445
      at next (/ti-mocha.js:6535:58013)
      at /ti-mocha.js:6535:57975
      at next (/ti-mocha.js:6535:57665)
      at /ti-mocha.js:6535:57759</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Layout</td><td>TIMOB-23372 #6</td><td>5.005</td><td><pre>Error: timeout of 5000ms exceeded
at Titanium.&lt;anonymous&gt; (/ti-mocha.js:6535:53723)</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.WebView</td><td>data</td><td>30.025</td><td><pre>Error: timeout of 30000ms exceeded
at Titanium.&lt;anonymous&gt; (/ti-mocha.js:6535:53723)</pre></td></tr>
<tr><td>android.emulator.fs</td><td>#statSync() returns stats for this file</td><td>0.001</td><td><pre>Error: expected 0 to be above 0
at Assertion.fail (/node_modules/should/cjs/should.js:275:13)
      at Assertion.value (/node_modules/should/cjs/should.js:356:9)
      at Context.&lt;anonymous&gt; (/fs.addontest.js:977:22)
      at Test.Runnable.run (/ti-mocha.js:6535:54163)
      at Runner.runTest (/ti-mocha.js:6535:58604)
      at /ti-mocha.js:6535:59445
      at next (/ti-mocha.js:6535:58013)
      at /ti-mocha.js:6535:57975
      at next (/ti-mocha.js:6535:57665)
      at /ti-mocha.js:6535:57759</pre></td></tr>
<tr><td>ios.Titanium.Blob</td><td>.file</td><td>0.001</td><td><pre>Error: undefined is not an object (evaluating 'file.toString')
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.blob.test.js:207:12
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Blob</td><td>.nativePath</td><td>0</td><td><pre>Error: undefined is not an object (evaluating 'blob.nativePath.length')
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.blob.test.js:124:23
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem</td><td>openStream()</td><td>0.001</td><td><pre>Error: The iOS reported an error
[native code]
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.test.js:104:32
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>#getDirectoryListing() returns null for file</td><td>0.001</td><td><pre>Error: expected Array [] to not exist
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:739:17
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>#copy() copies File successfully to new path</td><td>0.003</td><td><pre>Error: -[TiFilesystemBlobProxy copyWithZone:]: unrecognized selector sent to instance 0x7f8c73642ee0
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:559:12
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>#copy() is a function</td><td>0.027</td><td><pre>Error: -[TiFilesystemBlobProxy copyWithZone:]: unrecognized selector sent to instance 0x7f8c73409e90
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:550:12
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>#modifiedAt() returns a Date</td><td>0</td><td><pre>Error: file.modifiedAt is not a function. (In 'file.modifiedAt()', 'file.modifiedAt' is undefined)
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:233:25
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>#createdAt() returns a Date</td><td>0</td><td><pre>Error: file.createdAt is not a function. (In 'file.createdAt()', 'file.createdAt' is undefined)
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:188:27
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>#spaceAvailable()</td><td>0.001</td><td><pre>Error: expected false to be above 0
fail@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/node_modules/should/cjs/should.js:275:19
value@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/node_modules/should/cjs/should.js:356:13
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:543:23
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>#open(MODE_READ)</td><td>0.001</td><td><pre>Error: newFile.open is not a function. (In 'newFile.open(Ti.Filesystem.MODE_READ)', 'newFile.open' is undefined)
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:528:20
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>.size</td><td>0.001</td><td><pre>Error: expected 0 to be above 0
fail@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/node_modules/should/cjs/should.js:275:19
value@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/node_modules/should/cjs/should.js:356:13
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:91:27
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.Filesystem.File</td><td>.parent</td><td>0.001</td><td><pre>Error: expected 'Ti.Filesystem.File' to have a read-only property with name: parent
    expected Object {} to have property `parent` with descriptor {"writable":false}
        expected 'Unable to find property parent up the prototype chain!' undefined undefined
fail@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/node_modules/should/cjs/should.js:275:19
value@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/node_modules/should/cjs/should.js:356:13
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti.filesystem.file.test.js:83:37
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:54386
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58826
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:59671
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58234
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:58198
next@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57886
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:57982
timeslice@file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:5764:23</pre></td></tr>
<tr><td>ios.Titanium.UI.TableView</td><td>Delete row (Search Active)</td><td>6.37</td><td><pre>Error: timeout of 5000ms exceeded
file:///Users/build/Library/Developer/CoreSimulator/Devices/E1914280-4286-4FB6-9D85-C732FE9BCF33/data/Containers/Bundle/Application/103CBD32-6625-41D8-8DE3-C11CF109199D/mocha.app/ti-mocha.js:6535:53951</pre></td></tr>
</table>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d55ffb6bb30e192e552f7a3acef2b75d449c58f6
</p>

sgtcoolguy committed:
feat(ios): improve api compatibility for encrypted files

- add createdAt(), return Date equivalent of 0 timestamp
- add modifiedAt(), return Date equivalent of 0 timestamp
- make getDirectoryListing() return null
- attempt to have spaceAvailable() return a value by asking parent dir's proxy to retrieve it
- allow open() with TI_READ
- implement copy() (with encrypted file as source)
- implement parent property
- implement size property

sgtcoolguy committed:
test(file): if deployType==='test' no timestamps of encrypted files

sgtcoolguy commented:
I've pushed a couple follow-on commits to address some of the test failures once we get past the crash. Much like on Android, iOS uses a special class to handle encrypted files (TiFilesystemBlobProxy, equivalent to Android's TiResourceFile). This implementation didn't have some methods at all, also didn't have implementations of some we could feasibly do. I fixed those issues.

Haven't looked at the Blob failures yet, I assume something similar there.

sgtcoolguy committed:
refactor(ios): Ti.Filesystem.openStream calls getFile().open()

sgtcoolguy committed:
build: allow passing along deploy type to test command

sgtcoolguy committed:
ci(jenkins): use deploy type of test for unit test suite

sgtcoolguy committed:
fix(ios): retain filepath for encrypted file blobs

garymathews commented on iphone/Classes/FilesystemModule.m:
```diff
@@ -171,16 +180,21 @@ - (JSValue *)getFile
-    NSURL *url = [NSURL fileURLWithPath:newpath];
+- (TiFile *)getFileProxy:(NSString *)path
+{
+  if ([path hasPrefix:[self resourcesDirectory]] && ([path hasSuffix:@".html"] || [path hasSuffix:@".js"] || [path hasSuffix:@".css"] || [path hasSuffix:@".json"])) {
```
I believe we can remove the suffix checks for `html`, `css` since we don't encrypt those.

sgtcoolguy committed:
fix(ios): don't check for encrypted html or css files, should be none

sgtcoolguy committed:
Merge branch 'master' into TIMOB-27515

garymathews reviewed (approved):
CR: **PASS**

lokeshchdhry commented:
FR Passed.

`Ti.Filesystem.openStream` does not cause a crash with `deploytype` test.

Studio Ver:       5.1.4.201909061933
SDK Ver:          8.3.0 local build
OS Ver:           10.14.5
Xcode Ver:        Xcode 11.1
Appc NPM:         4.2.15
Appc CLI:         7.1.2-master.3
Daemon Ver:       1.1.3
Ti CLI Ver:       5.2.2
Alloy Ver:        1.14.1
Node Ver:         12.13.0
NPM Ver:          6.12.0
Java Ver:         10.0.2
Simulator:        Iphone 11 pro 13.1
